### PR TITLE
Initial Prescription swagger doc

### PR DESCRIPTION
**Pull request by mphprogrammer**
Adding swagger documentation to api

### CODE REVIEW

**Comment by leannammiller** [comment]
@mphprogrammer will estimate when able

**Comment by mphprogrammer** [comment]
@leannammiller I would like @aub to take a peek at the what I have so far (I've never used swagger or swagger-blocks before), if it is fundamentally correct, I'll have it done today.

**Comment by leannammiller** [comment]
@aub ^^

**Comment by saneshark** [comment]
@aub I'm wondering what your thoughts are on this one. I noticed you are helping review the SM and that one is mostly blocked by attachments working, specs passing right now from what it looks like to me. I also can't help to think there might be some opportunity for refactoring some commonalities (but would maybe prefer to have that as a separate future PR). Thoughts?

Would like to get this one merged before SM if possible to help guide F/E on any work they might need to do with prescription tracking changes.

File: app/swagger/requests/prescriptions.rb [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+          parameter name: :page, in: :query, required: false, type: :integer,
+                    description: 'Page of results, greater than 0'
+
+          parameter name: :per_page, in: :query, required: false, type: :integer,
+                    description: 'number of results, between 1 and 99'
+
+          parameter name: :sort, in: :query, required: false, type: :string,
+                    description: "Comma separated sort field(s), prepend field(s) with '-' for descending sort"
+
+          parameter name: :filter, in: :query, required: false, type: :string,
```
Comment (saneshark) [inline app/swagger/requests/prescriptions.rb]: Some of these might end up being common components across a variety of models and could later be included in a more modular way?

File: app/swagger/requests/prescriptions.rb [hunk 2]
```diff
@@ -0,0 +1,306 @@
...
+          key :tags, %w(prescriptions)
+
+          parameter name: :page, in: :query, required: false, type: :integer,
+                    description: 'Page of results, greater than 0'
+
+          parameter name: :per_page, in: :query, required: false, type: :integer,
+                    description: 'number of results, between 1 and 99'
+
+          parameter name: :sort, in: :query, required: false, type: :string,
+                    description: "Comma separated sort field(s), prepend field(s) with '-' for descending sort"
```
Comment (saneshark) [inline app/swagger/requests/prescriptions.rb]: Seems like they are already being duplicated here. Is there an opportunity to consolidate these in some manner and DRY things up?

File: app/swagger/requests/prescriptions.rb [hunk 3]
```diff
@@ -0,0 +1,306 @@
...
+
+          parameter name: :id, in: :path, required: true, type: :integer, description: 'id of the presecription'
+
+          response 204 do
+            key :description, 'prescription refill response'
+          end
+
+          response 400 do
+            key :description, 'prescription not available'
+
```
Comment (saneshark) [inline app/swagger/requests/prescriptions.rb]: I actually think there is another type of error that could occur or "prescription not available" the only one?

File: app/swagger/requests/prescriptions.rb (hunk 2 above)
Comment (mphprogrammer) [inline app/swagger/requests/prescriptions.rb]: @saneshark Yes - thats why this PR is on hold - look at the SM PR.

File: app/swagger/requests/prescriptions.rb (hunk 3 above)
Comment (mphprogrammer) [inline app/swagger/requests/prescriptions.rb]: I'll check that

File: app/swagger/requests/prescriptions.rb (hunk 1 above)
Comment (mphprogrammer) [inline app/swagger/requests/prescriptions.rb]: @saneshark Yes - thats why this PR is on hold - look at the SM PR.

**Comment by mphprogrammer** [comment]
@saneshark @aub The swagger-sm branch is more modular - once that was done, I was going to swing back and use the schemas and params in that PR for this.

**Comment by mphprogrammer** [comment]
@saneshark @aub Please review the swagger-sm PR, that is the template I am looking to follow.

**Pull request merged**